packages/zoltan2/core/src/TpetraCrsColorer/Zoltan2_TpetraCrsColorer.hpp

```diff
@@ -323,11 +320,9 @@ TpetraCrsColorer<CrsMatrixType>::reconstructMatrixFitted(
       "Row map of the Jacobian graph is not locally fitted to the vector's map."
       "  You must call the non-fitted version of this function.");
 
-  W.sync_device();
-
-  auto W_view_dev = W.getLocalViewDevice();
-  auto local_matrix = mat.getLocalMatrix();
-  auto local_graph = graph->getLocalGraph();
+  auto W_view_dev = W.getLocalViewDevice(Tpetra::Access::ReadOnly);
+  auto local_matrix = mat.getLocalMatrixDevice();
+  auto local_graph = graph->getLocalGraphDevice();
```

Comment:
Technically `local_matrix.graph` is cheaper as it will not recreate the local graph